packages/msw-addon/tsup.config.ts

```diff
@@ -0,0 +1,51 @@
+import { defineConfig } from 'tsup'
+
+const browser = defineConfig({
+  entry: {
+    'index.browser': './src/index.new.ts',
+  },
+  outDir: './dist',
+  target: ['chrome112'],
+  format: 'cjs',
+  esbuildOptions(options) {
+    options.alias = {
+      ...(options.alias || {}),
+      '@build-time/initialize': './src/initialize.browser.ts',
+    }
+  },
+  bundle: true,
+})
+
+const node = defineConfig({
+  entry: {
+    'index.node': './src/index.new.ts',
+  },
+  outDir: './dist',
+  target: 'node18',
+  format: 'cjs',
+  esbuildOptions(options) {
+    options.alias = {
```

Comment:
We can technically keep a single `initialize.ts` and then use build-time condition via `define` to strip away any conditional imports. But I find it risky because we are opening the doors to accidentally mixing environments if we forget to check for them. 

Our functionality here is tiny. There's no harm in repeating.